### PR TITLE
Use .colorhighlighter file for color_variables_files

### PR DESCRIPTION
**Pull request by jonsuh**
## Problem
People have reported that `color_variables_files` does not seem to work in [issue 296](https://github.com/Monnoroch/ColorHighlighter/issues/296).

Currently, `color_variables_files` is required to be in `.sublime-project` but if you don’t save and open directories as a “Sublime project,” then it won’t read the contents of `.sublime-project` and ColorHighlighter doesn’t work.

## Solution
I think it would be a good idea to also allow people to use a `.colorhighlighter` file that’s structured the same as `.sublime-project`.

Create a `.colorhighlighter` file in the root of your project and add the following in the file:
```
{
  "color_variables_files": [
    "/absolute/path/of/project/assets/_variables.scss"
  ]
}
```
That way it doesn’t require people to open directories as a “Sublime project” and will still work when you open directories from the command line using the `subl` command.

## Need Help
I managed to get my changes to work, but I am a Python n00b; therefore, will need help from more experienced devs to refactor and improve the code.

### CODE REVIEW

**Comment by Romainpetit** [comment]
Works like a charm for me. The notion of "sublime project" is just noise: we only want to reference the variables files the easiest way possible.
**This deserves to be merged.**
As for the quality of code, I couldn't judge it.

**Comment by bgarrant** [comment]
Has this feature been added?  I am also having the issue where I need to open a project to get this working.  There must be a better way.  The Gutter Color icon works across less/sass files and no configuration is needed to get the gutter colors to work.

**Comment by Romainpetit** [comment]
@Monnoroch What's your position about this point ?

File: ColorHighlighter.py [hunk 1]
```diff
@@ -1241,6 +1242,37 @@ def get_vars(self, view):
...
+
+                        # If .colorhighlighter file file in path exists
+                        if os.path.isfile(sublime_project):
+                            # Read file
+                            with open(sublime_project) as contents:
+                                # Try to parse JSON
+                                try:
+                                    sublime_project_data = json.load(contents)
+                                except Exception:
+                                    pass
```
Comment (Monnoroch) [inline ColorHighlighter.py]: At least log it. People need to see what error have they made in the console.

File: ColorHighlighter.py [hunk 2]
```diff
@@ -1241,6 +1242,37 @@ def get_vars(self, view):
                     if color_vars_file is not None:
                         color_vars_files.append(color_vars_file)
 
+                    # If not opened as a project
+                    if wnd.project_file_name() is None:
+                        # Path of .colorhighlighter file based on folders path
+                        sublime_project = pdata.get("folders", {})[0]["path"] + "/.colorhighlighter"
```
Comment (Monnoroch) [inline ColorHighlighter.py]: This is not a sublime project. This variable should have a better name.

Comment (Monnoroch) [inline ColorHighlighter.py]: As well as the sublime_project_data and the sublime_project_data_var below.

File: ColorHighlighter.py [hunk 3]
```diff
@@ -1241,6 +1242,37 @@ def get_vars(self, view):
...
+                                # If "color_variables_files" key exists
+                                if "color_variables_files" in sublime_project_data:
+                                    sublime_project_data_var = sublime_project_data["color_variables_files"]
+                                    # If string
+                                    if type(sublime_project_data_var) is str:
+                                        color_vars_files.append(sublime_project_data_var)
+                                    # If list
+                                    elif type(sublime_project_data_var) is list:
+                                        # For each in list
+                                        for f in sublime_project_data_var:
```
Comment (Monnoroch) [inline ColorHighlighter.py]: `color_vars_files.extend(sublime_project_data_var)` instead of a loop.

File: ColorHighlighter.py [hunk 4]
```diff
@@ -1241,6 +1242,37 @@ def get_vars(self, view):
                     if color_vars_file is not None:
                         color_vars_files.append(color_vars_file)
 
+                    # If not opened as a project
```
Comment (Monnoroch) [inline ColorHighlighter.py]: This code should be put in a separate function `parse_dot_colorhighlighter(pdata, color_vars_files)`. I know the current code is by far not in a good state, but let's not make it any worse.

**Comment by Monnoroch** [comment]
Sorry for the long response and big thanks for taking some time to fix the problem! Just some minor readability issues for fix.

File: ColorHighlighter.py [hunk 5]
```diff
@@ -1241,6 +1242,37 @@ def get_vars(self, view):
...
+                            with open(sublime_project) as contents:
+                                # Try to parse JSON
+                                try:
+                                    sublime_project_data = json.load(contents)
+                                except Exception:
+                                    pass
+
+                            # If valid JSON
+                            if sublime_project_data is not None:
+                                # If "color_variables_files" key exists
```
Comment (Monnoroch) [inline ColorHighlighter.py]: This comments and the ones below are quite useless. Comments should be used to explain why something is done, not, what. What is done is pretty clear from the code.

**Comment by Monnoroch** [comment]
The new version doesn't support variables just yet. I will definitely take this PR into account when implementing variables for the new version thought. Thanks!

**Pull request closed without merging**